### PR TITLE
Code Insights: Fixes repositories query preview link

### PR DESCRIPTION
**Pull request by vovakulikov**
Fixes https://github.com/sourcegraph/sourcegraph/issues/46920

By default on the backend we use repositories query with archived:true forked:true and count:all, we disallow these three filters in user input. In this PR I added these three in the client link generation logic. We could use query field from the API but since this is async API and we don't want to disabled preview link every time we fetch repositories count and did this on the client. 

Also this PR fixes small copy problem for the repositories list field.

## Test plan
- Make sure that preview link uses all three filters that insight backend uses for gathering repositories by repo query

## App preview:

- [Web](https://sg-web-vk-fix-paper-cut-insight-repo-ui.onrender.com/search)

Check out the [client app preview documentation](https://docs.sourcegraph.com/dev/how-to/client_pr_previews) to learn more.

### CODE REVIEW

**Comment by sg-e2e-regression-test-bob** [comment]
## Bundle size report 📦

| Initial size | Total size | Async size | Modules |
| --- | --- | --- | --- |
| 0.00% (0.00 kb) | 0.00% (+0.04 kb) | 0.00% (+0.04 kb) | 0.00% (0) |

Look at the [Statoscope report](https://console.cloud.google.com/storage/browser/_details/sourcegraph_reports/statoscope-reports/vk/fix-paper-cut-insight-repo-ui/compare-report.html;tab=live_object) for a full comparison between the commits 1af37f6 and bfbf022 or [learn more](https://docs.sourcegraph.com/dev/how-to/testing#bundlesize).


<details>
<summary>Open explanation</summary>

- `Initial size` is the size of the initial bundle (the one that is loaded when you open the page)
- `Total size` is the size of the initial bundle + all the async loaded chunks
- `Async size` is the size of all the async loaded chunks
- `Modules` is the number of modules in the initial bundle
</details>

File: client/web/src/enterprise/insights/components/creation-ui/insight-repo-section/InsightRepoSection.tsx [hunk 1]
```diff
@@ -229,6 +229,7 @@ function SmartSearchQueryRepoField(props: SmartSearchQueryRepoFieldProps): React
     }
 
     const queryState = disabled ? EMPTY_QUERY_STATA : value
+    const previewQuery = value.query ? `${value.query} archived:yes fork:yes count:all` : value.query
```
Comment (leonore) [inline client/web/src/enterprise/insights/components/creation-ui/insight-repo-section/InsightRepoSection.tsx:232]: the issue with this fix is that this is actually incorrect for any query with multiple predicates (using `and` `or` `not)

take the query `repo:sourcegraph or repo:go`

* if we just append the filters to the query we get [480 results](https://sourcegraph.sourcegraph.com/search?q=context:global+repo:sourcegraph+or+repo:go+archived:yes+fork:yes+count:all&patternType=standard&sm=0&groupBy=repo)
* if we use the backend-generated query we used to run this we get [193 results](https://sourcegraph.sourcegraph.com/search?q=context:global+%28fork:yes+archived:yes+count:all+repo:sourcegraph/sourcegraph+OR+fork:yes+archived:yes+count:all+repo:go%29&patternType=standard&sm=0&groupBy=repo)

Comment (vovakulikov) [reply client/web/src/enterprise/insights/components/creation-ui/insight-repo-section/InsightRepoSection.tsx:232]: I think that archived, forked and count are global filters so we don't need to include them in each part of complex expression 

Meaning 
`(repo:sourcegraph/sourcegraph or repo:go) archived:yes fork:yes count:all` [search link](https://sourcegraph.sourcegraph.com/search?q=context:global+%28repo:sourcegraph/sourcegraph+or+repo:go%29+archived:yes+fork:yes+count:all&patternType=standard&sm=0&groupBy=repo)
is the same as 
`(fork:yes archived:yes count:all repo:sourcegraph/sourcegraph OR fork:yes archived:yes count:all repo:go)` [search link](https://sourcegraph.sourcegraph.com/search?q=context:global+%28fork:yes+archived:yes+count:all+repo:sourcegraph/sourcegraph+OR+fork:yes+archived:yes+count:all+repo:go%29&patternType=standard&sm=0&groupBy=repo)

Comment (vovakulikov) [reply client/web/src/enterprise/insights/components/creation-ui/insight-repo-section/InsightRepoSection.tsx:232]: So in this example I think it should be valid just use
```
const previewQuery = value.query ? `(${value.query}) archived:yes fork:yes count:all` : value.query
```

note that I added `()` for the `value.query` value

Comment (chwarwick) [reply client/web/src/enterprise/insights/components/creation-ui/insight-repo-section/InsightRepoSection.tsx:232]: I think that adding the `()` works here and has the same effect as moving the filters before query value

`archived:yes fork:yes count:all ${value.query}` 

When `or` is involved you can use those parameters multiple times in opposite ways for example:
```
archived:yes fork:yes repo:sourcegraph or repo:notsourcegraph archived:no fork:no  count:all
```

Lets just make sure the solution you put in the counts consistently come out the same as the api is returning.

Comment (vovakulikov) [reply client/web/src/enterprise/insights/components/creation-ui/insight-repo-section/InsightRepoSection.tsx:232]: @chwarwick if you check the example above where I put search queries, it seems like both query



`archived:yes fork:yes count:all (${value.query})`

and `archived:yes fork:yes repo:sourcegraph or repo:notsourcegraph archived:no fork:no  count:all`

returns the same result on the result page

Comment (chwarwick) [reply client/web/src/enterprise/insights/components/creation-ui/insight-repo-section/InsightRepoSection.tsx:232]: Ok and just to make sure we are talking about the same thing, my primary concern is that:

This 👇 
creation

consistently matches 👇 
preview

Comment (vovakulikov) [reply client/web/src/enterprise/insights/components/creation-ui/insight-repo-section/InsightRepoSection.tsx]: Oh yeah I haven't push this yet, I checked it now after I added () and it seems like it's working properly

Comment (vovakulikov) [reply client/web/src/enterprise/insights/components/creation-ui/insight-repo-section/InsightRepoSection.tsx]: You can try it now, I pushed the fix

**Review by chwarwick — APPROVED** [review]

**Pull request merged**